### DISCUSSION TITLE
Welcome to Adobe-Runtime-Support Discussions!

### DISCUSSION THREAD

**Discussion by PrimaryFeather** [airsdk/Adobe-Runtime-Support#584] [General]
## 👋 Welcome!

The "Discussions" section is a place to connect with other members of the AIR community. 
Don't be shy! Most of us are really nice and don't bite! :wink:

* Ask questions you're wondering about. (No need to open up issues for those).
* Share ideas and tell us what's important to you and how _you_ see the future of AIR.
* Engage with other community members.
* Welcome others and be friendly and open-minded! 

This forum specializes on the AIR SDK and runtime, of course. For library-related discussions, the [Starling Forum](https://forum.starling-framework.org) has proven to be a great starting point.

Cheers!

**Comment by marchbold** [comment]
Brilliant! I think this will serve us well being able to pull out discussions from actual AIR issues 👍

**Comment by teotigraphix** [comment]
I read and thought; there are discussions going back months. Then I realize what ma said, "pull OUT discussions from the issues".

This is what I do with my time. :)

**Reply by marchbold** [reply]
Yeah, I've moved a few of the "issues" across to here, but let me know if you see any others that shouldn't be in there. I don't have a lot of time to go through them all, but I think it's important for us to moderate them so we don't overwhelm harman :)

**Comment by Jason0247** [comment]
We have been looking forward to the support of HUAWEI Harmony OS, which may bring new opportunities to all developers and AIR development teams starting from 2024.

**Comment by BASICboy-RF** [comment]
Hey, I have a linux machine, and I have an application (Gacha Club for PC) that I use Wine to run. All of a sudden, it has stopped working, on the premise of exit code '-1', and another windows emulation program says that I don't have the correct version of adobe AIR even though I tried replacing it, moving it, changing permissions, and other. Any advice???

**Reply by 2jfw** [reply]
>    Hey, I have a linux machine, and I have an application (Gacha Club for PC) that I use Wine to run. All of a sudden, it has stopped working, on the premise of exit code '-1', and another windows emulation program says that I don't have the correct version of adobe AIR even though I tried replacing it, moving it, changing permissions, and other. Any advice???

This is a "Welcome" post/thread only and not the right "place" for asking questions or raising issues - I would suggest you to create a separate post for your concern. You should also likely provide additional information on this (e.g. versions of app/AIR runtime etc.) as the information you provided seem a bit too broad and unspecific ..

**Reply by BASICboy-RF** [reply]
Alright, thanks. I'll move it.

**Reply by Kat-AskingQuestions** [reply]
i have the exact same problem, where did this chat get moved to?

**Comment by wam-athota** [comment]
Hi, Is there a support phone number that we can reach out to for purchasing the Adobe AIR SDK and support license. We have sent email to Adobe Support <[Adobe.Support@harman.com](mailto:Adobe.Support@harman.com)> but there isnt any timely response. 

Thanks in advance!
~Amar

**Reply by ajwfrost** [reply]
We don't have a phone number but purchasing the license can be done via https://airsdk.harman.com/pricing
I'll ping my colleague to check your email too, sometimes things go into a spam folder..
thanks

**Reply by little-fat-girl** [reply]
Yeah, they are really slow, but it does get done.  If I had known it would take a month, I would have started the purchase WAY earlier.

**Comment by itisme2** [comment]
Hi everybody !
Does exist a list of all the last (3-4 Years) released versions of the air-runtime, so I download them and can find the right one for an old software that uses an old release of air ? Thanks in advance :)

**Reply by marchbold** [reply]
Try installing the AIR SDK Manager, it can download / install all the recent Harman releases.

https://airsdk.dev/docs/basics/getting-started

Or Harman have downloads linked through the release notes:

https://airsdk.harman.com/release_notes

**Reply by anfratharman** [reply]
Just to clarify - the AIR SDK Manager will provide versions of the AIR SDK, but if you're looking for older versions of the shared AIR runtime installers, we don't keep older ones available beyond what's on the website:
https://airsdk.harman.com/runtime

Old software should always work with one of the two versions there:
- 33.1.1.744 for any content that requires the internal WebKit capabilities, i.e. an HTML-based app or one that uses advanced features of HTMLLoader classes
- 50.2.4.1 or later for anything else... generally you should try this one first but if the content doesn't work on it, uninstall it and try the 33.1 version.

thanks

**Reply by marchbold** [reply]
(Apologies I missed the "runtime" part)

**Comment by SteveB46** [comment]
Hi! Thanks for creating this forum! I'm using Adobe Animate 2024 and AIR SDK 51.1 on Windows 10 and I'm trying to compile to a 64-bit Windows exe with AIR embedded. In the research I've done I've only been able to find people describe how to compile to 64-bit for Android (which I've done successfully), or to Windows using a different dev environment. I've tried addiing and moving the 64-bit "architecture" node (which this forum edited to just say "64" when typing the full node text, so I'm summarizing it) to every line my application.xml, but my app is always 32-bit. Any help would be appreciated. Thanks in advance!

**Reply by ajwfrost** [reply]
It should that putting `<architecture>64</architecture>` at the top level (within the `application` document) is enough:
https://airsdk.dev/docs/building/application-descriptor-files/elements/application#architecture
But it's possible that this would be ignored if Animate is actually passing a value in .. worth checking the troubleshooting output of the AIR SDK Manager when you run the packaging command to see how it's being called.
There is also an environment variable `AIR_WIN_ARCH` that you could set to `64` which I believe was intended for Flash Builder workarounds, but again, if Animate has already told ADT what architecture to use, I think that would also be ignored..

**Reply by SteveB46** [reply]
Thank you very much for your quick response, Andrew. I will try what you suggested.

**Reply by SteveB46** [reply]
Success! The combination of setting the environment variable and adding the architecture node in the application xml produced the 64 bit exe! Thank you so much Andrew! You rock sir!
